src/thunderbird_accounts/mail/clients.py

```diff
@@ -513,84 +424,129 @@ def get_dns_records(self, domain_name: str) -> list[dict]:
-                    data_content = '\n'.join(event_data)
-                    try:
-                        stages = json.loads(data_content)
-                        if isinstance(stages, list):
-                            for stage in stages:
-                                delivery_stages.append(stage)
-                                # Check for completion
-                                if isinstance(stage, dict) and stage.get('type') == 'Completed':
-                                    return delivery_stages
-                                elif isinstance(stage, str) and stage == 'Completed':
-                                    return delivery_stages
-                    except json.JSONDecodeError as e:
-                        logging.warning(f'Failed to parse SSE data: {e}')
-
-                # Reset for next event
-                event_data = []
-                current_event = None
-                continue
-
-            # Parse SSE fields
-            if line.startswith('event:'):
-                current_event = line[6:].strip()
-            elif line.startswith('data:'):
-                event_data.append(line[5:].strip())
-
-        print(delivery_stages)
-
-        # Parse the verification stages to determine success/failure
-        is_verified, critical_errors, warnings = self._parse_verification_stages(delivery_stages)
+        critical_errors = []
+        warnings = []
+
+        expected_host = settings.CONNECTION_INFO['SMTP']['HOST'].rstrip('.')
+
+        # 1. Check MX Records
+        try:
+            mx_answers = dns.resolver.resolve(domain_name, 'MX')
+            has_correct_mx = False
+            for rdata in mx_answers:
+                exchange = rdata.exchange.to_text().rstrip('.')
+                if exchange == expected_host:
+                    has_correct_mx = True
+                    break
+
+            if not has_correct_mx:
+                logging.warning(f'MX records found for {domain_name} but none match {expected_host}')
+                critical_errors.append('mxLookupError')
+
+        except (dns.resolver.NoAnswer, dns.resolver.NXDOMAIN, dns.resolver.NoNameservers):
+            critical_errors.append('mxLookupError')
+        except Exception as e:
+            logging.error(f'MX lookup failed for {domain_name}: {e}')
+            critical_errors.append('mxLookupError')
+
+        # 2. Check SPF Record
+        try:
+            txt_answers = dns.resolver.resolve(domain_name, 'TXT')
+            has_spf = False
+            expected_spf_include = f'include:spf.{expected_host}'
+
+            for rdata in txt_answers:
+                # rdata.strings is a list of bytes
+                txt_content = b''.join(rdata.strings).decode('utf-8')
+                if txt_content.startswith('v=spf1') and expected_spf_include in txt_content:
+                    has_spf = True
+                    break
+
+            if not has_spf:
+                warnings.append('spfRecordNotFound')
```

Comment:
We should probably define these as a enum or constant value since they're everywhere.